File: plugins/probo/models/box.py
from database.db import db
from flask_mongoengine import BaseQuerySet
import datetime
from slugify import slugify, Slugify, UniqueSlugify
from models.category import Category
from flask_mongoengine.wtf import model_form

# id = lambda: str(uuid.uuid4())


class Box(db.Document):
    iso = db.StringField(required=True)
    sort = db.IntField(required=False, default=0)
    name = db.StringField(required=True, unique=True)
    slug = db.StringField(required=False, unique=True)
    description = db.StringField(required=False, default='')
    media = db.ListField()
    sqm = db.BooleanField(required=True, unique=False, default=False)
    published = db.BooleanField(default=True)
    input_type = db.StringField(default='')
    created_at = db.DateTimeField(default=datetime.datetime.now, required=True)
    categories = db.ListField(db.ReferenceField(Category))

    meta = {
        'indexes': ['-created_at', 'slug', 'name', 'categories'],
        'ordering': ['-created_at', 'name'],
        'collection': 'dwd_boxes',
        'queryset_class': BaseQuerySet
    }

    def __setattr__(self, key, value):
        super(Box, self).__setattr__(key, value)
        if key == 'name':
            self.slug = slugify(self.name, to_lower=True)

File: microservices/find/models/Box.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;


const boxSchema = Schema({
    iso:{type: String, required: true},
    sort:{type: Number, required: false, default: 0},
    name:{type: String, required: true, unique: true},
    slug:{type: String, required: false, unique:true},
    description:{type: String, required: false, default: ''},
    media:{type: []},
    sqm:{type: Boolean, required: true, unique: false, default: false},
    published:{type: Boolean, default: true},
    input_type: {type: String, default: ''},
    created_at: {type: Date, required: true},
    // categories = db.ListField(db.ReferenceField(Category))
});


module.exports = Box = mongoose.model("boxes", boxSchema);
